backend/src/test/java/capstone/storage/backend/ItemServiceTest.java

```diff
@@ -0,0 +1,206 @@
+        //WHEN
+        Item actual = itemService.updateItem(itemToExpect);
+        Item expected = itemRepo.save(itemToExpect);
+        //THEN
+        assertEquals(expected, actual);
+    }
+
+    @Test
+    @DisplayName("method: existById -> should return true")
+    void existById() {
+        //GIVEN
+        String id = "123";
+        when(itemRepo.existsById(id)).thenReturn(true);
+        //WHEN
+        boolean actual = itemService.existById(id);
+        boolean expected = true;
+        //THEN
+        assertEquals(expected, actual);
+    }
+
+    @Test
+    @DisplayName("method: deleteItemById -> should give the itemToDelete to the itemRepo")
+    void deleteItemById() {
+        //GIVEN
+        Item itemToDelete = new Item("123",
+                "testName",
+                "testCategory",
+                "GER", "8710847909610",
+                "20",
+                "1");
+        //WHEN
+        doNothing().when(itemRepo).deleteById(itemToDelete.id());
+        itemService.deleteItemById(itemToDelete.id());
+        //THEN
+        verify(itemRepo).deleteById(itemToDelete.id());
+    }
+
+    @Test
+    @DisplayName("method : checkItemExisting -> should throw ItemAlreadyExistingException")
+    void ifItemIsAlreadyExistingThrowItemAlreadyExistingException() {
+        //GIVEN
+        String testEanToFind = "123";
+        AddItemDto testAddItemDto = new AddItemDto("8710847909610", testEanToFind, "1");
+        when(itemRepo.existsByItemNumber(testAddItemDto.itemNumber())).thenReturn(true);
+        //WHEN
+        try {
+            itemService.checkItemExisting(testAddItemDto, testEanToFind);
+            fail();
+        }
+        //THEN
+        catch (ItemAlreadyExistException e) {
+            String actual = e.getMessage();
+            String expected = ExceptionMessage.ITEM_ALREADY_EXIST_EXCEPTION_MESSAGE.toString();
+
+            assertEquals(expected, actual);
+        }
+    }
+
+    @Test
+    @DisplayName("method : checkItemExisting -> should return true")
+    void checkIfItemAlreadyExistWithExistingEanReturnTrue() {
+        //GIVEN
+        String testEanToFind = "123";
+        AddItemDto testAddItemDto = new AddItemDto("8710847909610", testEanToFind, "1");
+        when(itemRepo.existsByEan(testEanToFind)).thenReturn(true);
+        try {
+            itemService.checkItemExisting(testAddItemDto, testEanToFind);
+            fail();
+        }
+        //THEN
+        catch (ItemAlreadyExistException e) {
+            String actual = e.getMessage();
+            String expected = ExceptionMessage.ITEM_ALREADY_EXIST_EXCEPTION_MESSAGE.toString();
+
+            assertEquals(expected, actual);
+        }
+    }
+
+    @Test
+    @DisplayName("method : isNullOrEmpty -> should return true when all AddItemDto fields are null")
+    void checkIfIsNullOrEmptyReturnsTrueWhenAddITemDtoFieldsAreNull() {
+        //GIVEN
+        AddItemDto addItemDto = new AddItemDto(null, null, null);
+        //WHEn
```

Comment:
```suggestion
        //WHEN
```
(auch an weiteren Stellen)